sdk/src/Services/S3/Custom/Model/GetObjectRequest.cs

```diff
@@ -156,7 +156,7 @@ public ByteRange ByteRange
         // Check to see if ByteRange property is set
         internal bool IsSetByteRange()
         {
-            return this.byteRange != null;
+            return this.byteRange != null && this.byteRange?.FormattedByteRange != null;
```

Comment:
Nit, `byteRange?` seems unnecessary as null check is already done previously, I'd replace it with `this.byteRange.FormattedByteRange`.